### DISCUSSION TITLE
RFC8693 microservice

### DISCUSSION THREAD

**Discussion by corrideat** [ory/hydra#3128] [Show and tell]
I've made an RFC8693 (token exchange) microservice in TypeScript for Node.js. Currently, it has some hardcoded values in the claims section that make it require some changes to run in a general setup, but the plan is to make it more generic.

The code is at https://github.com/Exact-Realty/hydra-rfc8693/

Mostly, it trusts its inputs (with regard to claims in the exchanged token) and is meant to sit behind another service making requests to it, and not to be publicly available. Because of Hydra API limitations, issuing an exchanged token requires simulating some user interactivity (making and accepting login and consent requests).

**Comment by vinckr** [comment]
Hey @corrideat,
Awesome, thanks a ton for sharing!
Would you be open to add a short README, I would love to add it to [our community projects](https://www.ory.sh/docs/ecosystem/community).

**Reply by corrideat** [reply]
Hi @vinckr,

It took me a while (busy with other projects), but I've now had an opportunity to essentially rewrite it from scratch, making it suitable for a more general use (instead of having to modify the source for the user information) and added a README.

**Reply by vinckr** [reply]
Awesome! 
Will add it to the docs and also promote it a bit in our newsletter :)
Thanks 🙌

**Reply by corrideat** [reply]
Amazing!
